week-1-ts-ecosystem/w01e01-typescript-basics/src/utils/utils.ts

```diff
@@ -0,0 +1,17 @@
+import { User } from "../models/User";
+
+export function hasAddress(user: User): boolean {
+  return Boolean(user.address);
+}
+
+export function hasGivenAge(requiredAge: number): (user: User) => boolean {
+  return (user) => user.age >= requiredAge;
+}
+
+export function prepareText(
+  user: User,
+  isAdult: (user: User) => boolean
+): string {
+  return `User ${user.name} is ${isAdult(user) ? "adult" : "minor"}
+  and has${hasAddress(user) ? "" : " no"} address.`;
+}
```

Comment:
Ta nazwa jest zbyt ogólna, wpada tutaj jeszcze `prepareText` i to się miesza wszystko.

Nie byłbym taki oszczędny w plikach tutaj i dorzucił sobie to jako `user-validators.ts` albo jeśli bardzo chcesz `user-utlis.ts` i oddzielił tam te 2 funkcje + `isAdult`.

wtedy `user-validators.ts` można wrzucić do `models` bez niczego.

a `prepareText` dać jako `text-utlis.ts` ?